Editor/Utility/CinemachineSceneTools.cs

```diff
@@ -0,0 +1,564 @@
+        {
+            if (tools.ContainsKey(tool))
+            {
+                tools.Remove(tool);
+            }
+            
+            tools.Add(tool, new CinemachineSceneToolDelegates
+            {
+                ToggleSetter = toggleSetter,
+                IsDisplayedSetter = isDisplayedSetter,
+            });
+        }
+        
+        internal delegate bool ToolbarHandler();
+        static ToolbarHandler s_ToolBarIsDisplayed;
+        internal static void RegisterToolbarIsDisplayedHandler(ToolbarHandler handler)
+        {
+            s_ToolBarIsDisplayed = handler;
+        }
+
+        static bool s_ToolbarTurnOffByMe;
+        internal delegate bool ToolbarTurnOnOffHandler(bool on);
+        static ToolbarTurnOnOffHandler s_ToolBarDisplay;
+        internal static void RegisterToolbarDisplayHandler(ToolbarTurnOnOffHandler handler)
+        {
+            s_ToolBarDisplay = handler;
+        }
+
+        internal static void SetSolo(bool active)
+        {
+            if (active)
+            {
+                foreach (var o in Selection.gameObjects)
+                {
+                    var vcam = o.GetComponent<ICinemachineCamera>();
+                    if (vcam != null)
+                    {
+                        CinemachineBrain.SoloCamera = vcam;
+                        InspectorUtility.RepaintGameView();
+                        break;
+                    }
+                }
+            }
+            else
+            {
+                CinemachineBrain.SoloCamera = null;
+            }
+        }
+
+        internal static void SetTool(bool active, Type tool)
+        {
+            if (active)
+            {
+                s_ActiveExclusiveTool = tool;
+                EnsureCinemachineToolsAreExclusiveWithUnityTools();
+            }
+            else
+            {
+                s_ActiveExclusiveTool = s_ActiveExclusiveTool == tool ? null : s_ActiveExclusiveTool;
+            }
+        }
+
+        static void EnsureCinemachineToolsAreExclusiveWithUnityTools()
+        {
+            foreach (var toolHandle in s_ExclusiveTools)
+            {
+                toolHandle.Value.ToggleSetter(toolHandle.Key == s_ActiveExclusiveTool);
+            }
+            if (s_ActiveExclusiveTool != null)
+            {
+                Tools.current = Tool.None; // Cinemachine tools are exclusive with unity tools
+            }
+        }
+        
+        static CinemachineSceneToolUtility()
+        {
+            s_ExclusiveTools = new SortedDictionary<Type, CinemachineSceneToolDelegates>(Comparer<Type>.Create(
+                (x, y) => x.ToString().CompareTo(y.ToString())));
+            s_Tools = new SortedDictionary<Type, CinemachineSceneToolDelegates>(Comparer<Type>.Create(
+                (x, y) => x.ToString().CompareTo(y.ToString())));
+            s_RequiredTools = new SortedDictionary<Type, int>(Comparer<Type>.Create(
+                (x, y) => x.ToString().CompareTo(y.ToString())));
```

Comment:
Why are you using sorted dictionaries?  There will never be a large number of tools so it's probably less efficient than unsorted